code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -459,13 +459,13 @@
 		max_storage_space += I.get_storage_cost()
 
 //Useful for spilling the contents of containers all over the floor
-/obj/item/weapon/storage/proc/spill()
-	if (istype(loc, /turf))//If its not on the floor this might cause issues
-		var/turf/T = get_turf(src)
-		for (var/obj/O in contents)
-			contents.Remove(O)
-			O.forceMove(T)
-			O.tumble(2)
+/obj/item/weapon/storage/proc/spill(var/dist = 2, var/turf/T = null)
+	if (!T)//If its not on the floor this might cause issues

```

Comment:
actually i fixed the issues, it should work fine anywhere now. it uses get_turf so it'll always place things on a floor